src/averageAges.js

```diff
@@ -59,6 +88,37 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const filterChildren = people.filter((person) => {
+    if (person.mother === null) {
+      return false;
+    }
+
+    const motherExist = people.some((mother) => mother.name === person.mother);
+
+    if (onlyWithSon) {
+      return motherExist && person.sex === 'm';
+    }
+
+    return motherExist;
+  });
+
+  const ageDifference = filterChildren.map((child) => {
```

Comment:
As this is an array, make more suitable name
```suggestion
  const ageDifferences = filterChildren.map((child) => {
```